devicetypes/smartthings/fibaro-rgbw-controller.src/fibaro-rgbw-controller.groovy

```diff
@@ -26,562 +26,233 @@
-        def colorData = getColorData(value.colorName)
-        value.rh = colorData.rh
-        value.gh = colorData.gh
-        value.bh = colorData.bh
-        value.hex = "#${value.rh}${value.gh}${value.bh}"
-    }
-
-    if (( value.size() == 2) && (value.colorName) && (value.level)) { //being called from outside of device (App) with only color name and level
-		def colorData = getColorData(value.colorName)
-        value.rh = hex(colorData.r * value.level/100)
-        value.gh = hex(colorData.g * value.level/100)
-        value.bh = hex(colorData.b * value.level/100)
-        value.hex = "#${hex(colorData.r)}${hex(colorData.g)}${hex(colorData.b)}"
-    }
-
-    if (( value.size() == 3) && (value.red != null) && (value.green != null) && (value.blue != null)) { //being called from outside of device (App) with only color values (0-255)
-        value.rh = hex(value.red)
-        value.gh = hex(value.green)
-        value.bh = hex(value.blue)
-        value.hex = "#${value.rh}${value.gh}${value.bh}"
-    }
-
-    if (( value.size() == 4) && (value.red != null) && (value.green != null) && (value.blue != null) && (value.level)) { //being called from outside of device (App) with only color values (0-255) and level
-        value.rh = hex(value.red * value.level/100)
-        value.gh = hex(value.green * value.level/100)
-        value.bh = hex(value.blue * value.level/100)
-        value.hex = "#${hex(value.red)}${hex(value.green)}${hex(value.blue)}"
-    }
-
-    if(value.hue) {
-        sendEvent(name: "hue", value: value.hue, displayed: false)
-    }
-    if(value.saturation) {
-        sendEvent(name: "saturation", value: value.saturation, displayed: false)
-    }
-    if(value.hex?.trim()) {
-        sendEvent(name: "color", value: value.hex, displayed: false)
-    }
-    if (value.level) {
-        sendEvent(name: "level", value: value.level)
-    }
-    if (value.switch?.trim()) {
-        sendEvent(name: "switch", value: value.switch)
-    }
-
-    sendRGB(value.rh, value.gh, value.bh)
-}
-
-def setLevel(level, rate = null) {
-	log.debug "setLevel($level)"
-
-	if (level == 0) { off() }
-	else if (device.latestValue("switch") == "off") { on() }
-
-    def colorHex = device.latestValue("color")
-    if (colorHex == null)
-		colorHex = "#FFFFFF"
-
-    def c = hexToRgb(colorHex)
-
-    def r = hex(c.r * (level/100))
-    def g = hex(c.g * (level/100))
-    def b = hex(c.b * (level/100))
-
-	sendEvent(name: "level", value: level)
-    sendEvent(name: "setLevel", value: level, displayed: false)
-	sendRGB(r, g, b)
-}
-
-
-def setWhiteLevel(value) {
-	log.debug "setWhiteLevel: ${value}"
-    def level = Math.min(value as Integer, 99)
-    level = 255 * level/99 as Integer
-    def channel = 0
-
-	if (device.latestValue("switch") == "off") { on() }
+	def rgb
+	if (value.hex) {
+		rgb = value.hex.findAll(/[0-9a-fA-F]{2}/).collect { Integer.parseInt(it, 16) }
+	} else {
```

Comment:
should check for value.hue and value.saturation, I guess this if statement assumes it is going to be either one or the other, perhaps a better practice to remove that assumption